### PR TITLE
Feature-26-resource-transfer

### PR DESCRIPTION
**Pull request by ignatIgnatov**

### CODE REVIEW

File: src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@Data
+@RequiredArgsConstructor
+public class TransferResourceRequestDto {
+
+    private static final String QUANTITY_MIN_VALUE_MSG = "Quantity should not be less than 1.";
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java]: Quantity can be less than 1, e.g. 0.5. I see the min value is actually 0.

File: src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    private static final String QUANTITY_DECIMAL_PLACES_MSG =
+            "Quantity should not have more than 2 decimal places.";
+
+    @NotNull
+    private UUID ownerId;
+    @NotNull
+    private UUID recipientId;
+    @NotNull
+    private UUID resourceId;
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java]: Let's keep the names consistent between the Request and response. Either go for previousOwnerId, newOwnerId, transferredResourceid here or.. go for provider(Id), recipient(Id), resource(Id) in both. Can't decide which one is better. You can also come up with something different if you see fit. Just keep it consistent between the resource and response.

File: src/main/java/jewellery/inventory/service/ResourceInUserService.java [hunk 3]
```diff
@@ -118,4 +146,81 @@ private ResourceInUser createAndAddNewResourceInUser(
...
+
+    if (resourceInUserToReceive == null) {
+      ResourceInUser resourceInUser = createAndAddNewResourceInUser(transferResourceRequestDto, recipient, resourceToTransfer);
+      return resourceInUserRepository.save(resourceInUser);
+    } else {
+      resourceInUserToReceive.setQuantity(
+          resourceInUserToReceive.getQuantity() + transferResourceRequestDto.getQuantity());
+      return resourceInUserRepository.save(resourceInUserToReceive);
+    }
+  }
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/ResourceInUserService.java]: Let's just use `addResourceToUser` and `removeResourceFromUser`? Then everything below should be obsolete.

File: src/test/java/jewellery/inventory/helper/ResourceTestHelper.java [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 import java.util.UUID;
 import java.util.stream.Stream;
+
+import jewellery.inventory.dto.request.TransferResourceRequestDto;
```
Comment (VladoKat) [inline src/test/java/jewellery/inventory/helper/ResourceTestHelper.java]: Obsolete change. Do `git checkout origin/test src/test/java/jewellery/inventory/helper/ResourceTestHelper.java`

**Review by VladoKat — CHANGES_REQUESTED** [review]
You can leverage already implemented methods for adding and removing resources from/to user 😉. Also, pull and merge with origin/test to fix the conflicts that are currently showing.

**Comment by Alekseeybg** [comment]
Don't forget to resolve the merge conflicts ! :)

File: src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java (hunk 2 above)
Comment (Alekseeybg) [reply src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java]: Personally I like previousOwnerId, newOwnerId, transferredResourceid  better

File: src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java (hunk 1 above)
Comment (ignatIgnatov) [reply src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java]: Done!

File: src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java (hunk 2 above)
Comment (ignatIgnatov) [reply src/main/java/jewellery/inventory/dto/request/TransferResourceRequestDto.java]: Done!

File: src/main/java/jewellery/inventory/service/ResourceInUserService.java (hunk 3 above)
Comment (ignatIgnatov) [reply src/main/java/jewellery/inventory/service/ResourceInUserService.java]: Done!

File: src/test/java/jewellery/inventory/helper/ResourceTestHelper.java (hunk 4 above)
Comment (ignatIgnatov) [reply src/test/java/jewellery/inventory/helper/ResourceTestHelper.java]: Done!

File: src/test/java/jewellery/inventory/helper/ResourceTestHelper.java
Comment (VladoKat) [inline src/test/java/jewellery/inventory/helper/ResourceTestHelper.java:1]: I can still see this as a change. Have you tried what I recommended? We can take a look together.

**Review by VladoKat — APPROVED** [review]
LGTM

**Pull request merged**